### PR TITLE
Cleanup IDE warnings

### PR DESCRIPTION
**Pull request by doom369**
Motivation : 

IDE highlights a lot of warnings, code hard to read with them.

Modifications : 

Removed unnecessary explicit type arguments, `final` for private methods, replaces `for` with `for each`.

Result : 

Cleaner code.

### CODE REVIEW

File: client/src/main/java/org/asynchttpclient/cookie/CookieEncoder.java [hunk 1]
```diff
@@ -63,8 +63,7 @@ public static String encode(Collection<Cookie> cookies) {
         } else {
             Cookie[] cookiesSorted = cookies.toArray(new Cookie[cookies.size()]);
             Arrays.sort(cookiesSorted, COOKIE_COMPARATOR);
-            for (int i = 0; i < cookiesSorted.length; i++) {
-                Cookie cookie = cookiesSorted[i];
+            for (Cookie cookie : cookiesSorted) {
```
Comment (slandelle) [inline client/src/main/java/org/asynchttpclient/cookie/CookieEncoder.java]: Actually, I would tend to disagree here: the original form saves an Iterator. One could argue JIT escape analysis might be able to save it, but maybe not. Netty does the same thing.

**Comment by slandelle** [comment]
LGTM except for one small nit. Could you please revert this one single change?

File: client/src/main/java/org/asynchttpclient/cookie/CookieEncoder.java (hunk 1 above)
Comment (doom369) [inline client/src/main/java/org/asynchttpclient/cookie/CookieEncoder.java]: @slandelle what iterator do you mean? This is array, `for each` in arrays don't use iterator. Compiler extract this loop into your version, It has nothing to do with JIT. I can show you bytecode if you want :).

Comment (slandelle) [inline client/src/main/java/org/asynchttpclient/cookie/CookieEncoder.java]: Oh, right!

**Comment by slandelle** [comment]
Thanks!

**Pull request merged**